src/vs/platform/terminal/common/capabilities/capabilities.ts

```diff
@@ -92,6 +92,7 @@ export interface ICommandDetectionCapability {
 	readonly cwd: string | undefined;
 	readonly onCommandStarted: Event<ITerminalCommand>;
 	readonly onCommandFinished: Event<ITerminalCommand>;
+	readonly onCommandCleared: Event<ITerminalCommand[]>;
```

Comment:
Firing individual commands would be more convenient when we have more uses?

```ts
readonly onCommandCleared: Event<ITerminalCommand>;
```

Alternatively we could go with `Event<ITerminalCommand | ITerminalCommand[]>;` when we do that.

Reply:
yeah let's adjust it when there's a use case